src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy

```diff
@@ -128,4 +132,31 @@ class GitVersionPlugin implements Plugin<Project> {
         // look in parent directory
         return scanForRootGitDir(currentRoot.parentFile)
     }
+
+    private static String runGitCommand(File dir, String ...commands) {
+        List<String> cmdInput = new ArrayList<>()
+        cmdInput.add("git")
+        cmdInput.addAll(commands)
+        ProcessBuilder pb = new ProcessBuilder(cmdInput)
+        pb.directory(dir)
+        pb.redirectErrorStream(true)
+
+        Process process = pb.start()
+        BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()))
+
+        StringBuilder builder = new StringBuilder()
+        String line = null
+        while ((line = reader.readLine()) != null) {
+            builder.append(line)
+            builder.append(System.getProperty("line.separator"))
+        }
+
+        int exitCode = process.waitFor()
+        if (exitCode != 0) {
+            return ""
```

Comment:
Makes sense.

In that case, would you prefer that we simply throw an exception, or log a warning and fall back to JGit `git describe` behavior? Presume the former, but wanted to check.